analysis_driver/reader/demultiplexing_parsers.py

```diff
@@ -326,5 +326,37 @@ def parse_welldup_file(welldup_file):
                 in_summary = False
     return dup_per_lane
 
-
+def parse_adapter_trim_file(adapter_trim_file, run_id):
+    adapters_trimmed_by_id = {}
+    with open(adapter_trim_file) as open_file:
+        open_file = open_file.read()
+        open_file = (open_file.split('\n\n')[0]).split('\n')
+        for line in islice(open_file, 1, None):
+            lane = line.split()[0]
+            read = line.split()[1]
+            sample_id = line.split()[3]
+            trimmed_bases = line.split()[6]
+            id = (run_id, sample_id, lane)
+            if not adapters_trimmed_by_id.get(id):
+                adapters_trimmed_by_id[id] = {}
+            adapters_trimmed_by_id[id]['read_%s_trimmed_bases' % (read)] = int(trimmed_bases)
+    return adapters_trimmed_by_id
+
+
+
+def convert_barcode_from_run_sample_lane(adapters_trimmed_by_id, has_barcode):
+    run_element_adapters_trimmed = {}
+    for id in adapters_trimmed_by_id:
+        run_id, sample_id, lane = id
+        if has_barcode:
+            data = rest_communication.get_documents('run_elements',
```

Comment:
I don't think this will work as ```get_documents``` will return an array not a single element.
Moreover you make the assumption that the rest API has been populated already which might not be the case.
It would be better to search the barcodes_info list in the RunCrawler to find the right run_element